### PR TITLE
Fixed all warnings and refactor module structure

### PR DESCRIPTION
**Pull request by miratepuffin**
### What changes were proposed in this pull request?
In this PR we have:

- Fixed all warnings in the project
- Refactored the module structure as many modules (Core, DB, Python, etc) had 20 or more files at top level.
- Reformatted the whole project to adhere to our preferred style. This includes enabling crate level import granularity.

### Why are the changes needed?
The structure of the project was getting a little out of control.

### Does this PR introduce any user-facing change? If yes is this documented?
Some of the public API's will have moved, but this will be reflected in the documentation - all examples still work.

### How was this patch tested?
This has been tested with the CI (must to my pain as the refactoring tools in Intelij and VS-code make me sad)

### Are there any further changes required?
Many singular files are quite large and can be split up.

### CODE REVIEW

File: raphtory/src/core/tgraph/graph/tgraph.rs [hunk 1]
```diff
@@ -4,29 +4,24 @@ use dashmap::DashMap;
...
-    edge_store::{EdgeLayer, EdgeStore},
-    graph_props::GraphProps,
-    node_store::NodeStore,
-    props::Meta,
-    tgraph_storage::{GraphStorage, LockedIter},
-    timer::{MaxCounter, MinCounter, TimeCounterTrait},
-    vertex::{ArcEdge, ArcVertex, Vertex},
-    EID, VID,
-};
+use crate::core::storage::locked_view::LockedView;
```
Comment (fabianmurariu) [inline raphtory/src/core/tgraph/graph/tgraph.rs:7]: do you prefer this style of imports?

**Review by fabianmurariu — APPROVED** [review]
LGTM

File: raphtory/src/core/tgraph/graph/tgraph.rs (hunk 1 above)
Comment (miratepuffin) [reply raphtory/src/core/tgraph/graph/tgraph.rs:7]: This is just from cargo fmt - can always change it

File: examples/rust/src/bin/bench/main.rs [hunk 2]
```diff
@@ -1,8 +1,8 @@
 use raphtory::algorithms::pagerank::unweighted_page_rank;
-use raphtory::db::graph::Graph;
-use raphtory::db::mutation_api::AdditionOps;
-use raphtory::db::view_api::*;
+use raphtory::db::api::mutation::AdditionOps;
+use raphtory::db::api::view::*;
 use raphtory::graph_loader::source::csv_loader::CsvLoader;
+use raphtory::prelude::*;
```
Comment (ljeub-pometry) [inline examples/rust/src/bin/bench/main.rs:5]: the prelude import should be enough?

File: examples/rust/src/bin/btc/main.rs [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 use std::fs::File;
 use std::io::{prelude::*, BufReader, LineWriter};
 use std::time::Instant;
 
-use raphtory::db::graph::Graph;
-use raphtory::db::mutation_api::AdditionOps;
-use raphtory::db::view_api::*;
+use raphtory::db::api::mutation::AdditionOps;
+use raphtory::db::api::view::*;
+use raphtory::prelude::*;
```
Comment (ljeub-pometry) [inline examples/rust/src/bin/btc/main.rs:23]: Note that prelude is imported twice! The other two imports here should also already be included in the prelude

File: examples/rust/src/bin/hulongbay/main.rs [hunk 4]
```diff
@@ -8,10 +9,10 @@ use itertools::Itertools;
...
 use raphtory::core::Prop;
-use raphtory::db::graph::Graph;
-use raphtory::db::mutation_api::AdditionOps;
-use raphtory::db::view_api::*;
-use raphtory::db::view_api::{GraphViewOps, VertexViewOps};
+use raphtory::db::api::mutation::AdditionOps;
+use raphtory::db::api::view::*;
+use raphtory::db::api::view::{GraphViewOps, VertexViewOps};
+use raphtory::db::graph::graph::Graph;
 use raphtory::graph_loader::source::csv_loader::CsvLoader;
```
Comment (ljeub-pometry) [inline examples/rust/src/bin/hulongbay/main.rs:16]: just use prelude?

File: examples/rust/src/bin/lotr/main.rs [hunk 5]
```diff
@@ -1,11 +1,11 @@
 use itertools::Itertools;
 use raphtory::algorithms::temporal_reachability::temporally_reachable_nodes;
-use raphtory::core::utils;
+use raphtory::core::utils::hashing;
 use raphtory::core::Prop;
-use raphtory::db::graph::Graph;
-use raphtory::db::mutation_api::AdditionOps;
-use raphtory::db::view_api::*;
+use raphtory::db::api::mutation::AdditionOps;
+use raphtory::db::graph::graph::Graph;
```
Comment (ljeub-pometry) [inline examples/rust/src/bin/lotr/main.rs:6]: these are in the prelude, can be removed

File: raphtory-benchmark/benches/common/mod.rs [hunk 6]
```diff
@@ -1,9 +1,11 @@
+#![allow(dead_code)]
+
 use criterion::{measurement::WallTime, BatchSize, Bencher, BenchmarkGroup, BenchmarkId};
 use rand::seq::*;
 use rand::{distributions::Uniform, Rng};
-use raphtory::db::graph::Graph;
-use raphtory::db::mutation_api::AdditionOps;
-use raphtory::db::view_api::*;
+use raphtory::db::api::mutation::AdditionOps;
+use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory-benchmark/benches/common/mod.rs:7]: these are already in the prelude

File: raphtory-benchmark/benches/edge_add.rs [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
-use raphtory::core::vertex::InputVertex;
-use raphtory::db::graph::Graph;
+use raphtory::prelude::*;
 
 mod common;
 use rand::distributions::{Alphanumeric, DistString};
 use rand::{thread_rng, Rng};
-use raphtory::db::mutation_api::AdditionOps;
+use raphtory::core::tgraph::vertices::input_vertex::InputVertex;
+use raphtory::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory-benchmark/benches/edge_add.rs:8]: already in the prelude

File: raphtory-graphql/src/lib.rs [hunk 8]
```diff
@@ -14,10 +14,9 @@ mod graphql_test {
     use dynamic_graphql::dynamic::DynamicRequestExt;
     use dynamic_graphql::{App, FieldValue};
     use raphtory::core::Prop;
-    use raphtory::db::graph::Graph;
-    use raphtory::db::mutation_api::AdditionOps;
+    use raphtory::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory-graphql/src/lib.rs:17]: already in the prelude

File: raphtory/src/algorithms/local_clustering_coefficient.rs [hunk 9]
```diff
@@ -24,9 +24,9 @@
 //!
 //! ```rust
 //! use raphtory::algorithms::local_clustering_coefficient::{local_clustering_coefficient};
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::mutation_api::AdditionOps;
-//! use raphtory::db::view_api::*;
+//! use raphtory::db::api::mutation::AdditionOps;
+//! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/algorithms/local_clustering_coefficient.rs:28]: only prelude should be necessary

File: raphtory/src/algorithms/local_triangle_count.rs [hunk 10]
```diff
@@ -18,9 +18,9 @@
 //!
 //! ```rust
 //! use raphtory::algorithms::local_triangle_count::{local_triangle_count};
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::mutation_api::AdditionOps;
-//! use raphtory::db::view_api::*;
+//! use raphtory::db::api::mutation::AdditionOps;
+//! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/algorithms/local_triangle_count.rs:22]: only prelude should be necessary

File: raphtory/src/algorithms/reciprocity.rs [hunk 11]
```diff
@@ -23,9 +23,9 @@
 //!
 //! ```rust
 //! use raphtory::algorithms::reciprocity::{all_local_reciprocity, global_reciprocity};
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::mutation_api::AdditionOps;
-//! use raphtory::db::view_api::*;
+//! use raphtory::db::api::mutation::AdditionOps;
+//! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/algorithms/reciprocity.rs:27]: only prelude should be necessary

File: raphtory/src/algorithms/triangle_count.rs [hunk 12]
```diff
@@ -21,9 +21,9 @@ use rustc_hash::FxHashSet;
 /// # Example
 /// ```rust
 /// use std::{cmp::Reverse, iter::once};
-/// use raphtory::db::graph::Graph;
 /// use raphtory::algorithms::triangle_count::triangle_count;
-/// use raphtory::db::mutation_api::AdditionOps;
+/// use raphtory::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/algorithms/triangle_count.rs:25]: this should already be in the prelude

File: raphtory/src/algorithms/triplet_count.rs [hunk 13]
```diff
@@ -55,10 +55,10 @@ use crate::db::view_api::{GraphViewOps, VertexViewOps};
 /// # Example
 ///
 /// ```rust
-/// use raphtory::db::graph::Graph;
 /// use raphtory::algorithms::triplet_count::triplet_count;
-/// use raphtory::db::mutation_api::AdditionOps;
-/// use raphtory::db::view_api::*;
+/// use raphtory::db::api::mutation::AdditionOps;
+/// use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/algorithms/triplet_count.rs:60]: only prelude necessary

File: raphtory/src/core/storage/locked_view.rs [hunk 14]
```diff
@@ -3,7 +3,7 @@
 //! When a raphtory graph is created, the code will automatically shard the graph depending
 //! on how many shards you set the graph to originally have when initializing it.
 //!
-//! For example, Graph::new(4) will create a graph with 4 shards.
+//! For example, Graph::new() will create a graph with 4 shards.
```
Comment (ljeub-pometry) [inline raphtory/src/core/storage/locked_view.rs:6]: This text is very wrong now

File: raphtory/src/core/tgraph/edges/edge.rs [hunk 15]
```diff
@@ -3,21 +3,16 @@ use std::{
...
-    tgraph_storage::{GraphEntry, LockedGraphStorage},
-    vertex::Vertex,
-    GraphItem, VRef, EID, VID,
-};
+use crate::core::storage::locked_view::LockedView;
+use crate::core::storage::timeindex::{TimeIndex, TimeIndexOps};
+use crate::core::tgraph::properties::tprop::TProp;
+use crate::core::tgraph::vertices::vertex::Vertex;
+use crate::core::{storage::Entry, Direction, Prop};
+
```
Comment (ljeub-pometry) [inline raphtory/src/core/tgraph/edges/edge.rs:11]: remove blank line

File: raphtory/src/core/tgraph/edges/edge_store.rs [hunk 16]
```diff
@@ -3,11 +3,14 @@ use std::ops::{Deref, DerefMut, Range};
...
 
-use crate::core::{
-    edge_ref::EdgeRef, errors::MutateGraphError, timeindex::TimeIndex, tprop::TProp, Prop,
-};
-
-use super::{props::Props, EID, VID};
+use crate::core::storage::timeindex::TimeIndex;
+use crate::core::tgraph::properties::tprop::TProp;
+use crate::core::Prop;
+
```
Comment (ljeub-pometry) [inline raphtory/src/core/tgraph/edges/edge_store.rs:9]: remove blank line

File: raphtory/src/core/tgraph/graph/tgraph.rs [hunk 17]
```diff
@@ -4,29 +4,24 @@ use dashmap::DashMap;
...
+use crate::core::tgraph::edges::edge::EdgeView;
+use crate::core::tgraph::edges::edge_store::{EdgeLayer, EdgeStore};
+use crate::core::tgraph::properties::graph_props::GraphProps;
+use crate::core::tgraph::properties::tprop::TProp;
+use crate::core::tgraph::vertices::input_vertex::InputVertex;
+use crate::core::tgraph::vertices::vertex::{ArcEdge, ArcVertex, Vertex};
+use crate::core::tgraph::vertices::vertex_ref::VertexRef;
+use crate::core::tgraph::vertices::vertex_store::VertexStore;
+use crate::core::{storage::Entry, Direction, Prop, PropUnwrap};
+
```
Comment (ljeub-pometry) [inline raphtory/src/core/tgraph/graph/tgraph.rs:18]: remove blank line

File: raphtory/src/core/tgraph/graph/tgraph.rs (hunk 1 above)
Comment (ljeub-pometry) [reply raphtory/src/core/tgraph/graph/tgraph.rs:7]: I think it doesn't like the nested `{}`, I think you can still do `crate::core::{}` as long as you don't nest any imports with `{}` inside of it? Maybe?

File: raphtory/src/core/tgraph/properties/props.rs [hunk 18]
```diff
@@ -5,17 +5,15 @@ use std::{
...
-use crate::core::{
-    errors::{IllegalMutate, MutateGraphError},
-    lazy_vec::LazyVec,
-    tprop::TProp,
-    Prop,
-};
+use crate::core::storage::lazy_vec::LazyVec;
 
-use super::tgraph::FxDashMap;
+use crate::core::tgraph::graph::tgraph::FxDashMap;
```
Comment (ljeub-pometry) [inline raphtory/src/core/tgraph/properties/props.rs:10]: what is up with this import?

File: raphtory/src/core/tgraph/properties/graph_props.rs [hunk 19]
```diff
@@ -1,8 +1,11 @@
 use serde::{Deserialize, Serialize};
 
-use crate::core::{locked_view::LockedView, tprop::TProp, Prop};
+use crate::core::storage::locked_view::LockedView;
+use crate::core::tgraph::properties::tprop::TProp;
+use crate::core::Prop;
 
-use super::{props::DictMapper, tgraph::FxDashMap};
+use crate::core::tgraph::graph::tgraph::FxDashMap;
```
Comment (ljeub-pometry) [inline raphtory/src/core/tgraph/properties/graph_props.rs:7]: recursive import? can this be simplified?

File: raphtory/src/core/utils/hashing.rs [hunk 20]
```diff
@@ -1,3 +1,5 @@
+#![allow(dead_code)]
+
```
Comment (ljeub-pometry) [inline raphtory/src/core/utils/hashing.rs:2]: why? should we remove the dead code instead or does it seem useful in the future?

File: raphtory/src/db/api/view/internal/into_dynamic.rs [hunk 21]
```diff
@@ -37,23 +37,23 @@ impl<G: GraphViewOps> IntoDynamic for VertexSubgraph<G> {
...
         }
-        impl<G: $crate::db::view_api::GraphViewOps> GraphTag
+        impl<G: $crate::db::view::GraphViewOps> GraphTag
             for $crate::db::subgraph_vertex::VertexSubgraph<G>
         {
         }
-        impl GraphTag for $crate::db::view_api::internal::DynamicGraph {}
+        impl GraphTag for $crate::db::view::internal::DynamicGraph {}
     };
 }
```
Comment (ljeub-pometry) [inline raphtory/src/db/api/view/internal/into_dynamic.rs:59]: I added this but I think it shouldn't be used anymore? It is not useful now that python is in the core crate...

File: raphtory/src/db/graph/edge.rs [hunk 22]
```diff
@@ -5,13 +5,13 @@
...
-use crate::db::graph_window::WindowedGraph;
-use crate::db::vertex::VertexView;
-use crate::db::view_api::*;
+use crate::core::tgraph::edges::edge_ref::EdgeRef;
+use crate::core::tgraph::vertices::vertex_ref::VertexRef;
+use crate::core::utils::time::IntoTime;
+use crate::db::api::view::{BoxedIter, EdgeViewInternalOps};
+use crate::db::graph::vertex::VertexView;
+use crate::db::graph::views::window_graph::WindowedGraph;
+use crate::prelude::{EdgeListOps, EdgeViewOps, GraphViewOps, Prop, TimeOps, VertexViewOps};
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/edge.rs:14]: just import `*`?

File: raphtory/src/db/graph/graph.rs [hunk 23]
```diff
@@ -7,9 +7,9 @@
 //! # Examples
 //!
 //! ```rust
-//! use raphtory::db::view_api::*;
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::mutation_api::AdditionOps;
+//! use raphtory::db::api::view::*;
+//! use raphtory::db::api::mutation::AdditionOps;
+//! use raphtory::prelude::Graph;
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/graph.rs:12]: just import `prelude::*` and remove the other imports

File: raphtory/src/db/graph/graph.rs [hunk 24]
```diff
@@ -18,16 +18,16 @@
...
 use crate::prelude::{EdgeListOps, EdgeViewOps, GraphViewOps, VertexViewOps};
 
 use serde::{Deserialize, Serialize};
 use std::fmt::{Display, Formatter};
 use std::{path::Path, sync::Arc};
 
-use super::mutation_api::internal::{InheritAdditionOps, InheritPropertyAdditionOps};
-use super::view_api::internal::{Base, DynamicGraph, InheritViewOps, IntoDynamic};
+use crate::db::api::mutation::internal::{InheritAdditionOps, InheritPropertyAdditionOps};
+use crate::db::api::view::internal::{Base, DynamicGraph, InheritViewOps, IntoDynamic};
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/graph.rs:30]: remove the blank lines between imports to get proper sorting...

File: raphtory/src/db/graph/path.rs [hunk 25]
```diff
@@ -1,13 +1,14 @@
...
-use crate::db::view_api::*;
+use crate::core::utils::time::IntoTime;
+use crate::core::Direction;
+use crate::db::api::view::internal::GraphWindowOps;
+use crate::db::api::view::{BoxedIter, LayerOps};
+use crate::db::graph::edge::EdgeView;
+use crate::db::graph::vertex::VertexView;
+use crate::db::graph::views::window_graph::WindowedGraph;
+use crate::db::graph::views::layer_graph::LayeredGraph;
+use crate::prelude::{GraphViewOps, Prop, TimeOps, VertexListOps, VertexViewOps};
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/path.rs:11]: just import `*`?

File: raphtory/src/db/graph/vertex.rs [hunk 26]
```diff
@@ -1,15 +1,16 @@
...
-use crate::db::view_api::*;
+use crate::core::utils::time::IntoTime;
+use crate::core::Direction;
+use crate::db::api::view::internal::GraphPropertiesOps;
+use crate::db::api::view::{BoxedIter, LayerOps};
+use crate::db::graph::edge::{EdgeList, EdgeView};
+use crate::db::graph::path::{Operations, PathFromVertex};
+use crate::db::graph::views::window_graph::WindowedGraph;
+use crate::db::graph::views::layer_graph::LayeredGraph;
+use crate::prelude::{GraphViewOps, Prop, TimeOps, VertexListOps, VertexViewOps};
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/vertex.rs:13]: just import `*`?

File: raphtory/src/db/graph/vertices.rs [hunk 27]
```diff
@@ -1,12 +1,13 @@
...
+use crate::core::tgraph::vertices::vertex_ref::VertexRef;
+use crate::core::utils::time::IntoTime;
+use crate::core::Direction;
+use crate::db::api::view::{BoxedIter, LayerOps};
+use crate::db::graph::edge::EdgeView;
+use crate::db::graph::path::{Operations, PathFromGraph};
+use crate::db::graph::vertex::VertexView;
+use crate::db::graph::views::window_graph::WindowedGraph;
+use crate::db::graph::views::layer_graph::LayeredGraph;
+use crate::prelude::{GraphViewOps, Prop, TimeOps, VertexListOps, VertexViewOps};
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/vertices.rs:10]: just import *?

File: raphtory/src/db/graph/views/deletion_graph.rs [hunk 28]
```diff
@@ -1,24 +1,25 @@
...
-use crate::core::errors::GraphError;
+use crate::core::storage::timeindex::TimeIndexOps;
+use crate::core::tgraph::edges::edge_ref::EdgeRef;
 use crate::core::tgraph::VID;
-use crate::core::timeindex::TimeIndexOps;
+use crate::core::utils::errors::GraphError;
 use crate::core::{Direction, Prop};
-use crate::db::mutation_api::internal::InheritMutationOps;
-use crate::db::view_api::internal::{
+
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/views/deletion_graph.rs:6]: remove blank line

File: raphtory/src/db/graph/views/deletion_graph.rs [hunk 29]
```diff
@@ -79,9 +80,9 @@ impl GraphWithDeletions {
     /// # Example
     ///
     /// ```no_run
-    /// use raphtory::db::graph::Graph;
     /// use std::fs::File;
-    /// use raphtory::db::mutation_api::AdditionOps;
+    /// use raphtory::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/views/deletion_graph.rs:84]: just prelude should be enough

File: raphtory/src/db/graph/views/window_graph.rs [hunk 30]
```diff
@@ -25,9 +25,9 @@
 //!
 //! ```rust
 //!
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::mutation_api::AdditionOps;
-//! use raphtory::db::view_api::*;
+//! use raphtory::prelude::*;
+//! use raphtory::db::api::mutation::AdditionOps;
+//! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/views/window_graph.rs:30]: just prelude should be enough

File: raphtory/src/db/graph/views/window_graph.rs [hunk 31]
```diff
@@ -413,9 +414,9 @@ impl<G: GraphViewOps> GraphOps for WindowedGraph<G> {
 /// # Examples
 ///
 /// ```rust
-/// use raphtory::db::graph::Graph;
-/// use raphtory::db::mutation_api::AdditionOps;
-/// use raphtory::db::view_api::*;
+/// use raphtory::db::api::mutation::AdditionOps;
+/// use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/views/window_graph.rs:418]: just prelude should be enough

File: raphtory/src/db/internal/core_ops.rs [hunk 32]
```diff
@@ -1,15 +1,11 @@
...
-    prelude::Prop,
-};
+use crate::core::storage::locked_view::LockedView;
+use crate::core::storage::timeindex::TimeIndex;
+use crate::core::tgraph::edges::edge_ref::EdgeRef;
+use crate::core::tgraph::graph::tgraph::InnerTemporalGraph;
+use crate::core::tgraph::properties::tprop::TProp;
+use crate::core::tgraph::vertices::vertex_ref::VertexRef;
+use crate::db::api::view::internal::CoreGraphOps;
+use crate::{core::tgraph::VID, prelude::Prop};
```
Comment (ljeub-pometry) [inline raphtory/src/db/internal/core_ops.rs:8]: could improve grouping

File: raphtory/src/db/task/task_runner.rs [hunk 33]
```diff
@@ -244,8 +242,8 @@ impl<G: GraphViewOps, CS: ComputeState> TaskRunner<G, CS> {
         // To allow the init step to cache stuff we will copy everything from cur_local_state to prev_local_state
         prev_local_state.clone_from_slice(&cur_local_state);
 
-        while !done && self.ctx.ss() < steps && tasks.len() > 0 {
-            (done, shard_state, global_state, cur_local_state) = self.run_task_list(
+        while !_done && self.ctx.ss() < steps && tasks.len() > 0 {
```
Comment (ljeub-pometry) [inline raphtory/src/db/task/task_runner.rs:245]: does this even work?

File: raphtory/src/graph_loader/example/company_house.rs [hunk 34]
```diff
@@ -1,8 +1,8 @@
 use crate::core::Prop;
-use crate::db::graph::Graph;
-use crate::db::mutation_api::{AdditionOps, PropertyAdditionOps};
-use crate::db::view_api::*;
+use crate::db::api::mutation::{AdditionOps, PropertyAdditionOps};
+use crate::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/company_house.rs:3]: prelude should be enough

File: raphtory/src/graph_loader/example/lotr_graph.rs [hunk 35]
```diff
@@ -15,18 +15,18 @@
...
 //! println!("The graph has {:?} vertices", graph.num_vertices());
 //! println!("The graph has {:?} edges", graph.num_edges());
 //! ```
-use crate::db::graph::Graph;
+use crate::prelude::*;
 
 use crate::graph_loader::fetch_file;
 
-use crate::db::mutation_api::AdditionOps;
+use crate::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/lotr_graph.rs:29]: prelude should be enough

File: raphtory/src/graph_loader/example/reddit_hyperlinks.rs [hunk 36]
```diff
@@ -40,9 +40,9 @@
 //! ```
 
 use crate::core::Prop;
-use crate::db::graph::Graph;
-use crate::db::mutation_api::AdditionOps;
+use crate::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/reddit_hyperlinks.rs:43]: prelude should be enough

File: raphtory/src/graph_loader/example/stable_coins.rs [hunk 37]
```diff
@@ -1,9 +1,9 @@
 use crate::core::Prop;
-use crate::db::graph::Graph;
-use crate::db::mutation_api::AdditionOps;
-use crate::db::view_api::GraphViewOps;
+use crate::db::api::mutation::AdditionOps;
+use crate::db::api::view::GraphViewOps;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/stable_coins.rs:3]: prelude should be enough

File: raphtory/src/graph_loader/example/sx_superuser_graph.rs [hunk 38]
```diff
@@ -38,18 +38,17 @@
...
 //! let graph = sx_superuser_graph().unwrap();
 //!
 //! println!("The graph has {:?} vertices", graph.num_vertices());
 //! println!("The graph has {:?} edges", graph.num_edges());
 //! ```
 
-use crate::db::graph::Graph;
-
-use crate::db::mutation_api::AdditionOps;
+use crate::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/sx_superuser_graph.rs:49]: prelude should be enough

File: raphtory/src/graph_loader/mod.rs [hunk 39]
```diff
@@ -9,8 +9,8 @@
 //! Load a pre-built graph
 //! ```rust
 //! use raphtory::algorithms::degree::average_degree;
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::view_api::*;
+//! use raphtory::prelude::*;
+//! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/mod.rs:13]: prelude should be enough

File: raphtory/src/graph_loader/mod.rs [hunk 40]
```diff
@@ -31,12 +31,12 @@
 //! Load a graph from csv
 //!
 //! ```no_run
-//! use raphtory::db::graph::Graph;
 //! use raphtory::core::Prop;
 //! use std::time::Instant;
 //! use serde::Deserialize;
-//! use raphtory::db::mutation_api::AdditionOps;
+//! use raphtory::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/mod.rs:37]: prelude should be enough

File: raphtory/src/graph_loader/mod.rs [hunk 41]
```diff
@@ -161,17 +161,13 @@ fn unzip_file(zip_file_path: &str, destination_path: &str) -> std::io::Result<()
...
-    use crate::db::mutation_api::AdditionOps;
-    use crate::{
-        core::{utils, Prop},
-        db::{
-            graph::Graph,
-            view_api::{GraphViewOps, TimeOps, VertexViewOps},
-        },
-    };
+    use crate::db::api::mutation::AdditionOps;
+    use crate::db::api::view::{GraphViewOps, TimeOps, VertexViewOps};
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/mod.rs:165]: prelude should be enough

File: raphtory/src/graph_loader/source/csv_loader.rs [hunk 42]
```diff
@@ -5,11 +5,11 @@
 //! use std::path::{Path, PathBuf};
 //! use regex::Regex;
 //! use raphtory::core::Prop;
-//! use raphtory::core::utils::calculate_hash;
+//! use raphtory::core::utils::hashing::calculate_hash;
 //! use raphtory::graph_loader::source::csv_loader::CsvLoader;
-//! use raphtory::db::graph::Graph;
-//! use raphtory::db::mutation_api::AdditionOps;
+//! use raphtory::db::api::mutation::AdditionOps;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/source/csv_loader.rs:10]: prelude should be enough

File: raphtory/src/graph_loader/source/csv_loader.rs [hunk 43]
```diff
@@ -403,10 +403,10 @@ impl CsvLoader {
 
 #[cfg(test)]
 mod csv_loader_test {
-    use crate::core::utils::calculate_hash;
+    use crate::core::utils::hashing::calculate_hash;
     use crate::core::Prop;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/source/csv_loader.rs:407]: prelude should be enough

File: raphtory/src/python/graph.rs [hunk 44]
```diff
@@ -4,10 +4,10 @@
 //! create windows, and query the graph with a variety of algorithms.
 //! It is a wrapper around a set of shards, which are the actual graph data structures.
 //! In Python, this class wraps around the rust graph.
-use crate::core::errors::GraphError;
+use crate::core::utils::errors::GraphError;
 use crate::core::Prop;
-use crate::db::graph::Graph;
-use crate::db::mutation_api::{AdditionOps, PropertyAdditionOps};
+use crate::db::api::mutation::{AdditionOps, PropertyAdditionOps};
```
Comment (ljeub-pometry) [inline raphtory/src/python/graph.rs:9]: prelude should be enough

**Review by ljeub-pometry — CHANGES_REQUESTED** [review]
May as well clean up the imports some more while we are here...

File: raphtory/src/graph_loader/example/lotr_graph.rs [hunk 45]
```diff
@@ -14,7 +14,7 @@
 //! Example:
 //! ```rust
 //! use raphtory::graph_loader::example::lotr_graph::lotr_graph;
-//! use raphtory::db::graph::Graph;
+//! use raphtory::prelude::*;
 //! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/lotr_graph.rs:18]: prelude should be enough

File: raphtory/src/graph_loader/example/reddit_hyperlinks.rs [hunk 46]
```diff
@@ -30,7 +30,7 @@
 //! Example:
 //! ```no_run
 //! use raphtory::graph_loader::example::reddit_hyperlinks::reddit_graph;
-//! use raphtory::db::graph::Graph;
+//! use raphtory::prelude::*;
 //! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/reddit_hyperlinks.rs:34]: prelude should be enough

File: raphtory/src/graph_loader/example/sx_superuser_graph.rs [hunk 47]
```diff
@@ -37,7 +37,7 @@
 //! Example:
 //! ```no_run
 //! use raphtory::graph_loader::example::sx_superuser_graph::sx_superuser_graph;
-//! use raphtory::db::graph::Graph;
+//! use raphtory::prelude::*;
 //! use raphtory::db::api::view::*;
```
Comment (ljeub-pometry) [inline raphtory/src/graph_loader/example/sx_superuser_graph.rs:41]: prelude should be enough

**Review by ljeub-pometry — DISMISSED** [review]

File: js-raphtory/src/graph/mod.rs [hunk 48]
```diff
@@ -1,21 +1,23 @@
...
-use raphtory::core::errors::GraphError;
+use raphtory::core::utils::errors::GraphError;
 use raphtory::core::Prop;
-use raphtory::db::graph::Graph as TGraph;
-use raphtory::db::graph_window::WindowedGraph;
-use raphtory::db::mutation_api::AdditionOps;
-use raphtory::db::view_api::internal::BoxableGraphView;
-use raphtory::db::view_api::GraphViewOps;
-use raphtory::db::view_api::TimeOps;
+use raphtory::db::api::mutation::AdditionOps;
```
Comment (fabianmurariu) [inline js-raphtory/src/graph/mod.rs:13]: another one for prelude

File: raphtory/src/core/utils/hashing.rs (hunk 20 above)
Comment (miratepuffin) [reply raphtory/src/core/utils/hashing.rs:2]: Its not actually dead_code - its used outside the lib and yeah are just useful functions

File: raphtory/src/db/api/view/internal/into_dynamic.rs (hunk 21 above)
Comment (miratepuffin) [reply raphtory/src/db/api/view/internal/into_dynamic.rs:59]: Removed

File: raphtory/src/db/task/task_runner.rs (hunk 33 above)
Comment (miratepuffin) [reply raphtory/src/db/task/task_runner.rs:245]: It does - it was just to get the warning to stop complaining about a variable which is not read (even though it is)

**Pull request merged**